File: aoc/2024/01/pt2.py
from aoc import read
from collections import Counter

def main():
    a_list = []
    def build():
        for a, b in read(lambda x: map(int, x.split())):
            a_list.append(a)
            yield b
    b_counter = Counter(build())
        
    return sum(el * b_counter.get(el, 0) for el in a_list)


if __name__ == '__main__':
    print(main())

File: aoc/y2024/d01/pt2.ts
import { Counter } from '../../index';


const aList = [];
const build = async function *() {
  for await (const line of console) {
    const [a, b] = line.split(/\s+/).map(x => parseInt(x, 10));
  
    aList.push(a);
    yield b;
  }  
};

(async () => {
  const bCount = await Counter(build());
  console.log(aList.reduce((acc, a) => {
    return acc + (a * (bCount[a] || 0));
  }, 0));
})();
